components/avatar/avatar.tsx

```diff
@@ -45,7 +45,7 @@ const Avatar: React.FC<AvatarProps> = React.memo(({
   return (
     <span className={`avatar ${className}`}>
       {!showText && <img className="avatar-img" src={src} {...props} />}
-      {showText && <span className="avatar-text">{safeText(text)}</span>}
+      {showText && <span className="avatar-text" {...props}>{safeText(text)}</span>}
```

Comment:
Type `React.ImgHTMLAttributes<any>` cannot be directly applied here.
Type `AvatarProps` should be a union type : `AvatarProps = AvatarImageProps | AvatarTextProps`

Reply:
@unix like this?
```ts
type AvatarImageProps = Props & typeof defaultProps & React.ImgHTMLAttributes<any>;
type AvatarTextProps = Props & typeof defaultProps & React.HTMLAttributes<any>;
export type AvatarProps = AvatarImageProps | AvatarTextProps;
```

Reply:
```ts
type NativeAttrs = React.ImgHTMLAttributes<any> | React.HTMLAttributes<any>

export type AvatarProps = Props & typeof defaultProps & NativeAttrs
```
This should work.

Reply:
@unix Unfortunately it doesn't seem to work.
But ``ImgHTMLAttributes  & HTMLAttributes`` does work.

Reply:
I'll check it ASAP.

Reply:
Fixed #33 .